trillian/ctfe/handlers.go

```diff
@@ -320,6 +320,18 @@ func parseBodyAsJSONChain(c *LogContext, r *http.Request) (ct.AddChainRequest, e
 	return req, nil
 }
 
+// appendUserCharge adds the specified user to the passed in ChargeTo and
+// and returns the result.
+// If the passed-in ChargeTo is nil, then a new one is created with the passed
+// in user and returned.
+func appendUserCharge(a *trillian.ChargeTo, user string) *trillian.ChargeTo {
```

Comment:
Grr, Github ate a large comment of mine -- let me try to recreate it.

I think things could be simpler if there was also a `func (c *LogContext) chargeUser(r *http.Request) *trillian.ChargeTo` function which could be used in-line in request structure initialization in most places.  The only special case would then be the addChainInternal case:
```go
	req := trillian.QueueLeavesRequest{
		LogId: c.logID,
		Leaves: []*trillian.LogLeaf{&leaf},
		ChargeTo: c.chargeUser(r),
	}
	if c.instanceOpts.CertificateQuotaUser != nil {
		// TODO(al): ignore pre-issuers? Probably doesn't matter
		for _, cert := range chain[1:] {
			req.ChargeTo = appendUserCharge(req.ChargeTo, c.instanceOpts.CertificateQuotaUser(cert))
		}
	}	
```